csrc/pf_inner.c

```diff
@@ -1062,28 +1062,42 @@ DBUG(("XX ah,m,l = 0x%8x,%8x,%8x - qh,l = 0x%8x,%8x\n", ah,am,al, qh,ql ));
         case ID_FILE_REPOSITION: /* ( ud fid -- ior ) */
             {
                 file_offset_t offset;
+                cell_t offsetHigh;
+                cell_t offsetLow;
                 FileID = (FileStream *) TOS;
-                offset = M_POP;
-                /* Avoid compiler warnings on Mac. */
-                offset = (sizeof(file_offset_t) > sizeof(cell_t))
-						? (offset << 8*sizeof(cell_t)) : 0 ;
-                offset += M_POP;
+                offsetHigh = M_POP;
+                offsetLow = M_POP;
+                /* We do not support double precision file offsets in pForth.
+                 * So check to make sure the high bits are not used.
+                 */
+                if (offsetHigh != 0)
+                {
+                    TOS = -3; /* TODO err num? */
+                    break;
+                }
+                offset = offsetLow;
                 TOS = sdSeekFile( FileID, offset, PF_SEEK_SET );
             }
             endcase;
 
         case ID_FILE_POSITION: /* ( fid -- ud ior ) */
```

Comment:
Might also want to address the code in `ID_FILE_SIZE` which does similar acts to support 64-bit file sizes. I only suggest that for consistency. It would seem odd that the file size could be > than 64-bit, but there would be no possible way to access it.

Reply:
I made ID_FILE_SIZE more consistent. Thanks. Note that a signed 64-bit file size would be 8388607 Terabytes. The largest hard drive I have ever used is 6 TB and it contained many files. ;-)
Most folks can get by with 32-bit file sizes in a 32-bit pForth. If they need more then they can use a 64-bit pForth and address 64-bit sized files. I think the reason the Forth standard uses double precision is so that 16-bit Forths can access 32-bit file sizes.

Reply:
I seriously need to spend more time pre-reading my comments. I meant to say 'could be > than 32-bit'. I can appreciate the limited use of a file > 64-bit 😄 

This looks good to me! Thanks!